docker-compose.yml

```diff
@@ -0,0 +1,59 @@
+version: '2.1'
+volumes:
+  resin-data: {}
+services:
+  balena-supervisor:
```

Comment:
Would this become the name of the container?

Reply:
yeah this is the most important part of this whole thing (the reason why we're adding it actually). Without this Supervisor builds create the container in the api with the default "main" name so we have to set it here.

Are you thinking that the current name is `balena_supervisor` not `balena-supervisor`. The sysd service uses a hyphen so I'd like to match them. What do you think @cywang117 @pipex ? do we use undercore or hypen ?

Reply:
Note that if deployed via balena as an app, the container name would actually be `<app_id>_<service_name>` so something like `123456_balena-supervisor`.

If building or running with normal docker-compose the container name would be `<project_name>_<service_name>` where project name is the name of the parent folder by default.

The systemd service is going to need to be updated either way because currently we directly start/stop the container by name.

Reply:
good points @klutchell. I'm slowly thinking that I don't have enough context for this PR and just did it because @pipex asked me to.. maybe someone else should take over ?